From e097692cd26763ba8724bef9e8420c22c07a4707 Mon Sep 17 00:00:00 2001
From: Tao Bao <tbao@google.com>
Date: Wed, 31 Jan 2018 12:18:52 -0800
Subject: [PATCH 16/20] releasetools: Move the AVB salt setup into
 common.LoadInfoDict().

We used to do this in add_img_to_target_files.AddImagesToTargetFiles(),
which didn't cover the path when calling from make_recovery_patch. As a
result, /system/bin/install-recovery.sh contains different SHA values
from the actual images

Test: Set up aosp_bullhead to use AVB. `m dist`, then run the following
      command to verify the generated install-recovery.sh.

  $ ./build/make/tools/releasetools/validate_target_files.py \
      out/dist/aosp_bullhead-target_files-eng.zip

(cherry picked from commit 12d87fc174d8c2f47a1f738f0d6514c2ceb4d241)
Change-Id: Id7be8fb17072252fcd4d08db2057b8c4af053376
---
 tools/releasetools/add_img_to_target_files.py | 11 -----------
 tools/releasetools/common.py                  | 18 +++++++++++++++---
 2 files changed, 15 insertions(+), 14 deletions(-)

diff --git a/tools/releasetools/add_img_to_target_files.py b/tools/releasetools/add_img_to_target_files.py
index fe4c9e070..940853f27 100755
--- a/tools/releasetools/add_img_to_target_files.py
+++ b/tools/releasetools/add_img_to_target_files.py
@@ -626,17 +626,6 @@ def AddImagesToTargetFiles(filename):
 
   has_recovery = (OPTIONS.info_dict.get("no_recovery") != "true")
 
-  if OPTIONS.info_dict.get("avb_enable") == "true":
-    fp = None
-    if "build.prop" in OPTIONS.info_dict:
-      build_prop = OPTIONS.info_dict["build.prop"]
-      if "ro.build.fingerprint" in build_prop:
-        fp = build_prop["ro.build.fingerprint"]
-      elif "ro.build.thumbprint" in build_prop:
-        fp = build_prop["ro.build.thumbprint"]
-    if fp:
-      OPTIONS.info_dict["avb_salt"] = hashlib.sha256(fp).hexdigest()
-
   # A map between partition names and their paths, which could be used when
   # generating AVB vbmeta image.
   partitions = dict()
diff --git a/tools/releasetools/common.py b/tools/releasetools/common.py
index 8a4cdff50..10afb86ed 100644
--- a/tools/releasetools/common.py
+++ b/tools/releasetools/common.py
@@ -31,12 +31,10 @@ import tempfile
 import threading
 import time
 import zipfile
+from hashlib import sha1, sha256
 
 import blockimgdiff
 
-from hashlib import sha1 as sha1
-
-
 class Options(object):
   def __init__(self):
     platform_search_path = {
@@ -258,6 +256,20 @@ def LoadInfoDict(input_file, input_dir=None):
     d["fstab"] = None
 
   d["build.prop"] = LoadBuildProp(read_helper)
+
+  # Set up the salt (based on fingerprint or thumbprint) that will be used when
+  # adding AVB footer.
+  if d.get("avb_enable") == "true":
+    fp = None
+    if "build.prop" in d:
+      build_prop = d["build.prop"]
+      if "ro.build.fingerprint" in build_prop:
+        fp = build_prop["ro.build.fingerprint"]
+      elif "ro.build.thumbprint" in build_prop:
+        fp = build_prop["ro.build.thumbprint"]
+    if fp:
+      d["avb_salt"] = sha256(fp).hexdigest()
+
   return d
 
 
-- 
2.17.1

